### ISSUE TITLE
Remove --no-format-pedantic flag from gcc builds

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#164]
Clang requires -Wno-format-pedantic so that it does not whine about printf format strings.

This flag is not understood properly by gcc, and breaks the build, so some reason, only on opt debug builds, as far as I can tell (though I have not tested all variants.)

Brendan has this patch on the Buildbots

```
diff --git a/build/standalone.gypi b/build/standalone.gypi
index 8a8128d..6e44392 100644
--- a/build/standalone.gypi
+++ b/build/standalone.gypi
@@ -201,8 +201,25 @@
         # things when their commandline changes). Nothing should ever read this
         # define.
         'defines': ['CR_CLANG_REVISION=<!(<(DEPTH)/tools/clang/scripts/update.sh --print-revision)'],
-        'cflags+': [
-          '-Wno-format-pedantic',
+        'conditions': [
+          ['host_clang==1', {
+            'target_conditions': [
+              ['_toolset=="host"', {
+                'cflags+': [
+                  '-Wno-format-pedantic',
+                 ],
+              }],
+           ],
+          }],
+          ['clang==1', {
+            'target_conditions': [
+              ['_toolset=="target"', {
+                'cflags+': [
+                  '-Wno-format-pedantic',
+                 ],
+              }],
+           ],
+          }],
         ],
       }],
     ],
```

I'd like to get something committed to fix this, though this looks a bit messy. Maybe one of you gyp experts can make this a bit cleaner?

If not, I'm ok with submitting this, and the Google team could tell us of there is a better way,

**Comment by paul99** [comment]
More details are in https://github.com/paul99/v8m-rb/issues/128, which is effectively a dup of this one, which I will now close.

**Comment by paul99** [comment]
Was fixed a while ago in d7f813b4 GYP: Don't pass -Wno-format-pedantic to GCC.

Ákos submitted Brendan's patch: https://codereview.chromium.org/1369273003

Closing.
